### PR TITLE
Remove demotivating uses of "just"

### PR DESCRIPTION
**Pull request by n2ygk**
per "[How Not to Demotivate Your Learners]( https://carpentries.github.io/instructor-training/08-motivation/index.html)" usages of the J-word (just) that might be perceived as demotivating have been removed.

I tried to remove "just" only in the demotivating sense of "simply". I also replaced "just" with "only" in one or two instances. I would appreciate feedback. Thanks for the opportunity to contribute.

### CODE REVIEW

**Comment by murraycadzow** [comment]
I think this improves the lessons and reduces the trivialisation (albeit unintended) of the concepts of the sentences, which could be demotivating for learners.

File: _episodes/07-errors.md [hunk 1]
```diff
@@ -394,10 +394,10 @@ One reason for receiving this error is that you specified an incorrect path to t
 For example,
 if I am currently in a folder called `myproject`,
 and I have a file in `myproject/writing/myfile.txt`,
-but I try to just open `myfile.txt`,
+but I try to open `myfile.txt`,
 this will fail.
 The correct path would be `writing/myfile.txt`.
-It is also possible (like with `NameError`) that you just made a typo.
+It is also possible (like with `NameError`) that you made a typo.
```
Comment (murraycadzow) [inline _episodes/07-errors.md]: I would suggest replacing "you made a typo" with "a typo has occurred" to minimise what could be interpreted as accusation.

**Comment by maxim-belkin** [comment]
Thank you for the pull request, @n2ygk! 🎉 
It actually sparked a discussion among maintainers of what constitutes a proper use of the word "just". I must admit that it does not demotivate me so I will review changes from the point of view of its meaning: 
a. We should not be using "just" to imply simplicity/easiness.
b. We should discuss whether we should keep it whenever it implies "only". I'm OK with changing it to "only" but then we need to discuss whether it actually improves the wording.
c. We should either keep it in sentences where its removal makes language harsher or rephase sentences/paragraphs entirely.

**Comment by tracykteal** [comment]
Thank you for this PR. It is important to modify how we use language when learners and others tell us that language is demotivating, regardless of whether it is demotivating to a particular individual. We have feedback that language here is being perceived as dismissive, and the ability to address those comments, to create and maintain high-quality, inclusive materials. So, let's move forward on incorporating these updates, as @maxim-belkin suggests in b), replacing 'just' with 'only' as appropriate to maintain clarity, and otherwise removing or substituting it, updating if the validity of the content is affected.

File: _episodes/07-errors.md (hunk 1 above)
Comment (n2ygk) [reply _episodes/07-errors.md]: In this context, "all" feels  similar to "just" in that it implies that it's easy -- which it might not be for a learner.

File: _episodes/01-numpy.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ weight_kg = 60
 {: .language-python}
 
 From now on, whenever we use `weight_kg`, Python will substitute the value we assigned to
-it. In essence, **a variable is just a name for a value**.
+it. In essence, **a variable is a name for a value**.
```
[the remark this answers is not in the document]
Comment (n2ygk) [reply _episodes/01-numpy.md]: Perhaps better to call it a "place holder for one of a set of possible values" or something along those lines. Given the audience is primarily scientists, I think they get what a variable is.  Maybe even drop the whole "In essence..." sentence.

[the remark this answers is not in the document]
Comment (maxim-belkin) [reply _episodes/01-numpy.md]: > Perhaps better to call it a "place holder for one of a set of possible values"

I don't think this phrasing is better as it raises such questions as "what set of possible values?", what is a set?, which possible values?, does placeholder mean something in Python?. Current explanation uses terminology that is well known ("name", "value") and that has been just discussed ("variable").

At this point, I would prefer to keep changes to a minimum.

>  Maybe even drop the whole "In essence..." sentence.

This sentence is the culmination of a very brief introduction to variables. It is important for those who do not know what a variable is before the workshop. I'd like to emphasize that our lesson is titled "**Introduction to** Programming with Python" and as recently as 5/29/2019 I taught this lesson and there were learners who did not know anything about programming. I would not say that this is typical, but there is nothing wrong with that and even this brief introduction is better than nothing. I am considering to propose to split this brief introduction into a separate episode but this would have to be done in a separate PR and we would have to discuss it with those who teach this lesson.

[the remark this answers is not in the document]
Comment (maxim-belkin) [reply _episodes/01-numpy.md]: How about
> Essentially, a variable is a name for a value.

File: _episodes/09-debugging.md [hunk 3]
```diff
@@ -206,7 +206,7 @@ and we're better able to give them the information they need to be useful.
 And speaking of help:
 if we can't find a bug in 10 minutes,
 we should *be humble* and ask for help.
-Just explaining the problem aloud is often useful,
+Explaining the problem aloud is often useful,
```
[the remark this answers is not in the document]
Comment (maxim-belkin) [reply _episodes/09-debugging.md]: How about:
> Frequently, to better understand the problem it helps to explain it to someone else.

File: _episodes/01-numpy.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (n2ygk) [reply _episodes/01-numpy.md]: additional commit(s) coming shortly

File: _episodes/07-errors.md [hunk 4]
```diff
@@ -18,7 +18,7 @@ keypoints:
    then it will be called an `IndentationError`."
 - "A `NameError` will occur if you use a variable that has not been defined,
    either because you meant to use quotes around a string, you forgot to define the variable,
-   or you just made a typo."
+   or you made a typo."
```
[the remark this answers is not in the document]
Comment (n2ygk) [reply _episodes/07-errors.md]: "... or you may have misspelled or used different capitalization of the variable name."

File: _episodes/07-errors.md [hunk 5]
```diff
@@ -270,7 +270,7 @@ NameError: name 'hello' is not defined
 ~~~
 {: .error}
 
-The second is that you just forgot to create the variable before using it.
+The second is that you forgot to create the variable before using it.
```
[the remark this answers is not in the document]
Comment (n2ygk) [reply _episodes/07-errors.md]: hmm, I'm actually OK with this usage; it was also used earlier.

File: _episodes/09-debugging.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (n2ygk) [reply _episodes/09-debugging.md]: added rubber ducking

File: _episodes/02-loop.md [hunk 6]
```diff
@@ -58,7 +58,7 @@ d
 This is a bad approach for three reasons:
 
 1.  **Not scalable**. Imagine you need to print characters of a string that is hundreds
-    of letters long.  It might be easier just to type them in manually.
+    of letters long.  It might be easier to type them in manually.
```
[the remark this answers is not in the document]
Comment (maxim-belkin) [reply _episodes/02-loop.md]: I'd like other maintainers to chime in here.

File: _episodes/02-loop.md [hunk 7]
```diff
@@ -165,7 +165,7 @@ of the loop body (e.g. `end for`); what is indented after the `for` statement be
 >
 > In the example above, the loop variable was given the name `char` as a mnemonic;
 > it is short for 'character'.
-> We can choose any name we want for variables. We might just as easily have chosen the name
+> We can choose any name we want for variables. We might as easily have chosen the name
```
[the remark this answers is not in the document]
Comment (maxim-belkin) [reply _episodes/02-loop.md]: Let's mark conversations resolved once all the changes are added

File: _episodes/07-errors.md [hunk 8]
```diff
@@ -97,9 +97,9 @@ the program encountered an error on Line 6, when it tried to run the code `print
 > Sometimes, you might see a traceback that is very long
 > -- sometimes they might even be 20 levels deep!
 > This can make it seem like something horrible happened,
-> but really it just means that your program called many functions before it ran into the error.
+> but really it means that your program called many functions before it ran into the error.
 > Most of the time,
-> you can just pay attention to the bottom-most level,
+> you can pay attention to the bottom-most level,
```
[the remark this answers is not in the document]
Comment (maxim-belkin) [reply _episodes/07-errors.md]: `only` is essential to the meaning here.

File: _episodes/07-errors.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (maxim-belkin) [reply _episodes/07-errors.md]: I'm not OK with `you forgot`

**Comment by n2ygk** [comment]
@maxim-belkin I've mostly incorporated the review feedback.

Commits are organized to facilitate cherry picking with the biggest change being that I felt
`data` was not a good example of a variable name as it could get confused with the generic meaning. However, this requires fixing one of the png diagrams.  See `TODO` inside comments in the source.

I also flagged the container diagram as not matching the code example.

File: _episodes/05-cond.md [hunk 9]
```diff
@@ -155,8 +155,8 @@ if maximum inflammation in the beginning (day 0) and in the middle (day 20) of
 the study are equal to the corresponding day numbers.
 
 ~~~
-max_inflammation_0 = numpy.max(data, axis=0)[0]
-max_inflammation_20 = numpy.max(data, axis=0)[20]
+max_inflammation_0 = numpy.max(inflamation, axis=0)[0]
```
Comment (maxim-belkin) [inline _episodes/05-cond.md]: Please undo this (and all other `data` to `inflamation` conversion) change: it has a typo and has not been discussed. We can discuss this change in a separate PR.

File: _episodes/07-errors.md [hunk 10]
```diff
@@ -97,10 +97,9 @@ the program encountered an error on Line 6, when it tried to run the code `print
 > Sometimes, you might see a traceback that is very long
 > -- sometimes they might even be 20 levels deep!
 > This can make it seem like something horrible happened,
-> but really it just means that your program called many functions before it ran into the error.
-> Most of the time,
-> you can just pay attention to the bottom-most level,
-> which is the actual place where the error occurred.
+> but all it means is that your program called many functions before it ran into the error.
+> Most of the time, the actual place where the error occurred is at the bottom-most level,
+> so just skip over the traceback down to the bottom line.
```
Comment (maxim-belkin) [inline _episodes/07-errors.md]: "just"

**Comment by n2ygk** [comment]
FYI - I'm going on vacation for 10 days so please don't take my silence as lack of follow through....

File: _episodes/02-loop.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (n2ygk) [reply _episodes/02-loop.md]: @maxim-belkin No other chimes. Marking as resolved. Feel free to re-review as you see fit.

File: _episodes/07-errors.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (n2ygk) [reply _episodes/07-errors.md]: @maxim-belkin suggestion on what to use? "you neglected"?

[the remark this answers is not in the document]
Comment (ldko) [reply _episodes/07-errors.md]: What about "The second is that you are trying to use a variable that hasn't been created yet." or "The second is that you are trying to use a variable that does not yet exist." @n2ygk @maxim-belkin

[the remark this answers is not in the document]
Comment (maxim-belkin) [reply _episodes/07-errors.md]: > What about "The second is that you are trying to use a variable that hasn't been created yet." or "The second is that you are trying to use a variable that does not yet exist."

I like both options! Thank you, @ldko!

**Comment by n2ygk** [comment]
@maxim-belkin: I think I'm done. Please approve as you see fit. Thanks.

File: _episodes/01-numpy.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (maxim-belkin) [reply _episodes/01-numpy.md]: My suggestions are:

> Basically, **a variable is a name for some value**.

> Basically, **a variable (`weight_kg`) is a name for a value (`60`)**.

> In layman's terms, **a variable is a name for a value**.

What do you think, @n2ygk, @brownsarahm, @ldko?

[the remark this answers is not in the document]
Comment (ldko) [reply _episodes/01-numpy.md]: Since we are looking at this closely, I think any of these could be considered misleading because the variable is not just the name, but the stored value with a name. This difference is highlighted with the next line `In Python, variable names` where "variable names" is used to indicate the name of a variable over just "variable". Because of that, my suggestion is to say something like:
```
From now on, wherever we use `weight_kg`, Python will substitute the value we assigned to it. Basically, a variable is a value that has a name.
```

[the remark this answers is not in the document]
Comment (brownsarahm) [reply _episodes/01-numpy.md]: This is a tricky one, because what it's doing is explicitly not addressing what actually happens and tries to suggest a heuristic as a mental model. I personally think that a name for a value is a bad mental model for a variable, because it suggests permanence. However, I agree that we want to introduce some heuristic and not get into actually how memory works. Maybe an appropriately simplified version is that a variable is a named container that we can put values in and change. 

```
One way to think of a variable is that it is a name for a container and when we refer to it, it's contents, a value is filled in by python. 
```

I think for the carpentries' target audience of researchers, the concept of a variable in algebra is reliable prior knowledge. So we could instead build from there. I think this approach is better, it avoids demotivating language and tries to use something learners are more familiar with to teach the new concept.  

```
Like a variable in a mathematical expression, a variable stands in for a value.  What's different is that variables in python can have a wide variety of values and the value of a variable can change. 
```

[the remark this answers is not in the document]
Comment (jttkim) [reply _episodes/01-numpy.md]: I prefer the explanation of a variable of a container. This extends well to further explaining that the value is the content of the container, but code can be developed without having that content, values are only required at run time.

I'm less sure about the mathermatical reference. I don't understand how variables taking a wide variety of values is "different" from a variable in a mathematical expression, as a mathematical variable can take any value in e.g. a function's domain. And more generally, it's the nature of a variable that the values it holds can vary.

[the remark this answers is not in the document]
Comment (ldko) [reply _episodes/01-numpy.md]: I like the track of a variable being described as a named container. I also would prefer to avoid bringing in the mathematical expression comparison.

[the remark this answers is not in the document]
Comment (maxim-belkin) [reply _episodes/01-numpy.md]: Thank you all, for your comments! This is indeed a tricky question/concept to explain. Here are my thoughts...

TL;DR

Let's discuss this sentence/paragraph/section in a separate PR/issue.

<hr>

> Basically, a variable is a value that has a name.

This definition emphasizes value over name and, therefore, is even more vulnerable to criticism of implying a static nature of value that @brownsarahm mentions as a critique of the current definition. I also think this definition changes the meaning of the sentence and of what a variable is. Here are some analogies:

<br>

(Current version): "In essence, Buddy (variable) is a nickname for a dog (value)."
(This suggestion): "Basically, Buddy (variable) is a dog (value) that has a nickname."

(Current version): "In essence, a nickname (variable) is a [short] name for a person (value)."
(This suggestion): "Basically, a nickname (variable) is a person (value) with a [short] name."

(Current version): "In essence, a laptop (variable) is a [short] name for a small computer (value)."
(This suggestion): "Basically, a laptop (variable) is a small computer (value) that has a [short] name."

<br>

<hr>

> the variable is not just the name, but the stored value with a name.

This is where it becomes tricky. Usually (and by saying "usually" I mean languages like C, Fortran), a variable is a human-readable name for an address in (and the required size of) computer memory. This memory address always holds some value:

> Human --> variable --> memory address (+ size) === value

`-->` means abstractions
`===` means equivalence.

So, usually, the connection between a variable and a value is through a memory address that does not change once a variable is created. For example, one can create an integer `int a = 0;` in C, check its address in memory, increment value by 1 with `a=a+1;`, and address of `a` in memory will be the same. In Python, even simple operations like `a=a+1` change memory address of a variable. Given that both value and memory address can change, variable's name is, in fact, the only thing that is permanent. That why making a mental connection between variable and its name is probably the right thing to do in Python.

What I don't "like" about this definition is the word "value": it is too abstract and it indeed might look as if we're giving a name to a specific value. Perhaps, "observable quantity" or "data" are better alternatives to "value".

<hr>

### Side notes

The most important thing about variables is that they are human-readable names and not something like `0x7ffee16c26fc`. So, while variables are certainly connected to values, they are not values themselves. 

In Python, variables cannot exist without values (`None` is also a value). In other languages they can: `int a;` in C declares `a` as a name for a variable of type integer. However, before setting its value with something like `a = 10;` the value of `a` is not defined -- it (the value) depends on what is located at the memory address that operating system gives to the program at the time someone asks about the value of `a`.

<hr>

> This is a tricky one, because what it's doing is explicitly not addressing what actually happens and tries to suggest a heuristic as a mental model. I personally think that a name for a value is a bad mental model for a variable, because it suggests permanence.

I agree that "a name for a value" is not the best definition but it's the word "value" that is not right here. See my notes above. How about, "observable quantity" or "data"?

<hr>

> Maybe an appropriately simplified version is that a variable is a named container that we can put values in and change.

> I prefer the explanation of a variable of a container. This extends well to further explaining that the value is the content of the container, but code can be developed without having that content, values are only required at run time.

> I like the track of a variable being described as a named container.

I don't mind introducing this mental model and, perhaps, we should. Note, however, that we already have one mental model -- variables as sticky notes -- which is not a bad one. Perhaps, we could extend / amend / clarify it. I like this definition of variables: https://launchschool.com/books/ruby/read/variables#whatisavariable

[the remark this answers is not in the document]
Comment (n2ygk) [reply _episodes/01-numpy.md]: Maybe I'll just borrow this definition: http://swcarpentry.github.io/python-novice-gapminder/02-variables/index.html. I'm not going anywhere near launchschool.com's definition as it is not CC licensed.

[the remark this answers is not in the document]
Comment (n2ygk) [reply _episodes/01-numpy.md]: ... which means leaving it as is.

File: _episodes/01-numpy.md [hunk 11]
```diff
@@ -266,7 +266,7 @@ when there's nothing interesting after the decimal point.
 Our call to `numpy.loadtxt` read our file
 but didn't save the data in memory.
 To do that,
-we need to assign the array to a variable. Just as we can assign a single value to a variable, we
```
[the remark this answers is not in the document]
Comment (n2ygk) [reply _episodes/01-numpy.md]: Thanks, but I believe "in a similar way" is just (:-) as good

File: _episodes/02-loop.md [hunk 12]
```diff
@@ -33,7 +33,7 @@ word = 'lead'
 ~~~
 {: .language-python}
 
-In Python, a string is just an ordered collection of characters, so every
+In Python, a string is an ordered collection of characters and every
```
[the remark this answers is not in the document]
Comment (n2ygk) [reply _episodes/02-loop.md]: I think it's helpful to indicate that it is an ordered collection as later, when we iterate over a string, list, tuple or other containers, it will make sense that it is a collection and that there's an ordering. 

I don't see how adding "basically"  changes the meaning.

File: _episodes/02-loop.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (n2ygk) [reply _episodes/02-loop.md]: Since you initially stated you are OK with the change, when inviting chimes, I'm going to stick with your first answer.

**Comment by n2ygk** [comment]
@maxim-belkin, @jttkim, @murraycadzow, @ldko, @brownsarahm: I defer to you and the other editors to merge/close/further change as you see fit. I've become demotivated (just kidding) about making further edits but can't commit any additional time to this PR. 

Let me know if you want me to rebase.

Thank you all for the opportunity to contribute and for your suggestions.

Cheers!

**Comment by ldko** [comment]
To help facilitate getting @n2ygk's contribution of removing demotivating language merged, I have made a few minor tweaks to language per the discussion. @maxim-belkin what do you think?

**Comment by maxim-belkin** [comment]
Thanks, @ldko! I would like to discuss some of the changes more, so I suggest to break this entire PR into several small ones -- a change per PR -- and discuss them separately. 

I can help you with this process if you're willing to do that.

**Comment by ldko** [comment]
I think at this point in the review process (already significant conversation, changes requested, additional changes made), splitting this PR into one per change (from what I count that would be 22 PRs) is perhaps not worth the benefits (and loses sight of discussion contributed by others). To me the current changes remove language that some learners might find demotivating, address the suggestions/comments you have previously made, and don't significantly change the meaning of the lesson. If @maxim-belkin @annefou want to comment further on this PR, I'll do what I can to address the comments.

**Comment by maxim-belkin** [comment]
I'm not comfortable with any of us pushing to Alan's fork as he refused to continue working on this PR.

**Comment by n2ygk** [comment]
It seems you don't really want the changes, which is JUST fine. I don't
quite understand the governance around these lessons but someone needs to
make a decision at this point.

I've also learned that our particular SWC group is not teaching with this
lesson but rather gapminder, so I am not motivated to continue endless
discussions about a lesson that is seeing reduced use -- and none by me.

**Comment by n2ygk** [comment]
@ldko Feel free to push to my fork as you see fit; It's not going anywhere. If you are more comfortable forking mine, that's fine too. Let me know how I can help.

File: _episodes/07-errors.md [hunk 13]
```diff
@@ -270,7 +270,7 @@ NameError: name 'hello' is not defined
 ~~~
 {: .error}
 
-The second is that you just forgot to create the variable before using it.
+The second is that you are trying to use a variable that does not yet exist.
```
Comment (maxim-belkin) [inline _episodes/07-errors.md]: Let's add "reason" after "The second"

Comment (maxim-belkin) [reply _episodes/07-errors.md]: "you are trying" -> "you might be trying"

File: _episodes/07-errors.md [hunk 14]
```diff
@@ -394,10 +394,10 @@ One reason for receiving this error is that you specified an incorrect path to t
 For example,
 if I am currently in a folder called `myproject`,
 and I have a file in `myproject/writing/myfile.txt`,
-but I try to just open `myfile.txt`,
+but I try to open `myfile.txt`,
 this will fail.
 The correct path would be `writing/myfile.txt`.
-It is also possible (like with `NameError`) that you just made a typo.
+It is also possible (like with `NameError`) that the file path contains a typo.
```
Comment (maxim-belkin) [inline _episodes/07-errors.md]: Move ```(like with `NameError`)``` to the end of the sentence, remove parentheses so that it reads something like this:

> It is also possible that the file name contains a typo in which case Python returns a `NameError`.

I'm not sure if we can you file _path_

File: _episodes/09-debugging.md [hunk 15]
```diff
@@ -121,7 +121,7 @@ are almost never worthwhile.
 
 If it takes 20 minutes for the bug to surface,
 we can only do three experiments an hour.
-That doesn't just mean we'll get less data in more time:
+That doesn't only mean we'll get less data in more time:
```
Comment (maxim-belkin) [inline _episodes/09-debugging.md]: We need to improve this sentence -- "This doesn't only mean" is hard to parse. Simplest fix:

> This means that we'll get less date in more time and that we're more likely to be distracted...

but alternative solutions are also welcome.

File: _episodes/07-errors.md [hunk 16]
```diff
@@ -97,10 +97,10 @@ the program encountered an error on Line 6, when it tried to run the code `print
 > Sometimes, you might see a traceback that is very long
 > -- sometimes they might even be 20 levels deep!
 > This can make it seem like something horrible happened,
-> but really it just means that your program called many functions before it ran into the error.
-> Most of the time,
-> you can just pay attention to the bottom-most level,
-> which is the actual place where the error occurred.
+> but the length does not reflect severity, rather,
```
Comment (maxim-belkin) [inline _episodes/07-errors.md]: "the length" is not clear... perhaps, "the length of the error message".

**Comment by maxim-belkin** [comment]
Thank you for working on this PR, Lauren. I left a few comments -- please address them (or respond with your thoughts) whenever you have time. This PR is getting very close to being complete.

I should also clarify that we should use `Co-authored-by` when the named co-author has made a significant contribution to the discussion of the change that you're making and approves of it. This is fine in this case because a) Alan is a first-time contributor and b) he explicitly allowed it; but you should also consider that you've spent your time doing that and haven't received (and won't receive) anything for that. After all, your contribution to the lesson is as important as anyone else's.

File: _episodes/07-errors.md (hunk 14 above)
Comment (annefou) [reply _episodes/07-errors.md]: file path looks weird to me too but I do not know if using file name is precise enough (as here we have the path too).

**Comment by ldko** [comment]
Thanks @maxim-belkin and @annefou for the feedback. All of the changes make sense to me to incorporate. I will try to get to making the changes today.

File: _episodes/07-errors.md (hunk 14 above)
Comment (ldko) [reply _episodes/07-errors.md]: Regarding incorporating `NameError` without parentheses, if the file name contains a typo, it does not return a `NameError`, it is still a `FileNotFoundError`. I could drop the `NameError` reference for:
```
It is also possible that the file name or its path contains a typo.
```

Comment (maxim-belkin) [reply _episodes/07-errors.md]: right... `NameError` is when there is a typo in a variable name. You can either drop these parentheses altogether or replace `NameError` with `FileNotFoundError` -- your call
